.github/workflows/ci.yml

```diff
@@ -0,0 +1,39 @@
+name: CI
+
+# Trigger the workflow on push or pull request, but only for the master branch
+on:
+  pull_request:
+  push:
+    branches: [master]
```

Comment:
can we trigger CI on all branches?

Reply:
Currently it will run the build for all branches that have the PR (pointing at master) open. Just to double-check: do we want to run builds for all branches, no matter if there's a PR open or not?

Reply:
if that's the case, why has it not run for the current PR? Do I need to enable something at the repo level?

Reply:
I'm not sure, but I think the GitHub actions config needs to exist on `master` branch. Let me try creating a PR from my fork's `master` branch to see if that's going to trigger it.

Reply:
Hmmm that didn't work. I think my changes need to be merged for the GitHub Actions to be enabled on this repo (but not sure). I'll read up on it more now, hopefully I'll figure something out.

Reply:
It looks like the configuration needs to exist in the target repo (which means GitHub Actions will start working after we merge this PR). See builds that get triggered in my fork: https://github.com/mjarosie/odd-jobs/pull/2

I'm going to get rid of the build trigger for `push` event as it duplicates builds (see the screenshot below). As far as I understand, the goal is to build the code whenever there's a PR to be sure that we're not merging broken code into `master`. Action triggered on the `pull_request` event should suffice.

![image](https://user-images.githubusercontent.com/9082353/94857350-46a01a80-0429-11eb-8886-4e8391d721f1.png)

Reply:
I've restricted the operating systems to Ubuntu only as Windows fails for some reason with the following message:

```
Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for direct-daemonize-3.1:
    unix must match >=2.4.0.1, but the stack configuration has no specified version  (latest
         matching version is 2.7.2.2)
needed due to odd-jobs-0.2.2 -> direct-daemonize-3.1

In the dependencies for odd-jobs-0.2.2:
    unix needed, but the stack configuration has no specified version  (latest matching version
         is 2.7.2.2)
needed since odd-jobs is a build target.

Some different approaches to resolving this:

  * Recommended action: try adding the following to your extra-deps
    in D:\a\odd-jobs\odd-jobs\stack.yaml:

- unix-2.7.2.2@sha256:9e93f93cc5a065248120136e83a0a6d1ce93d6eb5ef2a2543f9593e93e164d24,3496

Plan construction failed.
Error: Process completed with exit code 1.
```

It looks like making it build on Windows should be a separate task.

Reply:
> It looks like the configuration needs to exist in the target repo (which means GitHub Actions will start working after we merge this PR).

Okay - let me merge this and see what happens.